apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts

```diff
@@ -290,267 +362,306 @@ export function buildMachine(
       events: {} as PaperHandlerStatusEvent,
     },
     id: 'bmd',
-    initial: 'not_accepting_paper',
+    initial: 'voting_flow',
     context: initialContext,
     on: {
-      PAPER_JAM: 'jammed',
-      JAMMED_STATUS_NO_PAPER: 'jam_physically_cleared',
-    },
-    states: {
-      // Initial state. Doesn't accept paper and transitions away when the frontend says it's ready to accept
-      not_accepting_paper: {
-        invoke: pollPaperStatus(),
-        on: {
-          // Paper may be inside the machine from previous testing or machine failure. We should eject
-          // the paper (not to ballot bin) because we don't know whether the page has been printed.
-          PAPER_INSIDE_NO_JAM: 'eject_to_front',
-          PAPER_PARKED: 'eject_to_front',
-          BEGIN_ACCEPTING_PAPER: 'accepting_paper',
-        },
+      PAPER_JAM: 'voting_flow.jammed',
+      JAMMED_STATUS_NO_PAPER: 'voting_flow.jam_physically_cleared',
+      PAT_DEVICE_CONNECTED: {
```

Comment:
Interesting, I hadn't imagined this being part of the state machine. I had kind of assumed it would just be another separate backend API similar to how we interface with the card reader for auth. But I can see a case for it being integrated here if we want to interrupt the voting flow when the PAT device is connected but not calibrated. It's interesting to try to figure out these new patterns as we move more stuff to the backend.

That makes me think - do we need to interrupt the voting flow to calibrate the PAT device? How/when do we imagine this device getting set up and calibrated? With the accessible controller, it seems like we have a flow where it gets calibrated from a menu as needed, not in real time when connected.

Reply:
Great questions! You're correct that the main reason I made this part of the state machine was to support the ability to interrupt at any time.

The [product brief](https://docs.google.com/document/d/1lYqziJK7cPovJlWDthYJZiIPQAVv4UTZ4wPoWcomy8w/edit#heading=h.ktowgi2il3p) specifies that we should interrupt the voting flow to calibrate the device as soon as it's plugged in, and "At any point in the voter session, a user can connect a dual-switch device into the PAT port".

It's worth questioning whether this requirement in the product brief is necessary. I think it's a nice-to-have, since a voter or poll worker might forget to connect the device until they hand off control to the voter, or the voter might decide halfway through that the physical dimensions of the screen and elements require them to use a PAT device.

There's also some benefit to waiting until the cardless voter session is active (as opposed to having the poll worker set it up, like the accessible controller). The accessible controller test seems to be purely to test that the hardware works, whereas the PAT device calibration is primarily to teach the voter how to navigate the DOM in this particular app. Since the voter must be involved in that flow, I don't think we want to start the calibration flow from a menu.

Let me know if that makes sense; happy to discuss more.

Reply:
Thanks for sharing that context! I hadn't read the product brief before. I agree with your interpretation that this is primarily about teaching the voter.

Given that, it feel like this flow should be associated with the user session. E.g. if two voters in a row use the PAT device without unplugging it, both should see the PAT device tutorial. Is that how it behaves now? My read is no - that it only shows once per connection of the device.

Reply:
I've been assuming each voter will supply their own device ie. they will disconnect their device and take it with them after casting the ballot. But now that you mention it, I think there's also the possibility that the precinct supplies the device, so we should delete `context.patDevice` on session reset as well.

Reply:
https://github.com/votingworks/vxsuite/pull/3997/commits/72fb195a1989f9c38a6841b43121ca0fde59ddbd#diff-ea4f2566cf950ca374305ad00616d5eda7f645377dcf5f5cf6167e1fd0add195R631 resets `context.patDevice` and transitively the calibration flow after every voter session

Reply:
The main difference between the accessible controller input and dual-switch input is that we know the exact controls on the former, but don't necessarily know the latter. There's the possibility that some jurisdictions will provide dual-switch inputs at the polling place (though this is rare) so we have to support a flow that allows any dual-switch input. Even in the case where dual-switches are provided, a voter may still want to bring an alternative form that they are more comfortable with into the polling place and I'd like to be able to accommodate that.

All this is to say: I think we need to have a mapping/calibration step for each voter session, which is how I understand it to be currently implemented?

Reply:
Correct, it's per voter session right now. I think the question during review was whether we needed to support

a. connection at _any_ time during the voter flow vs
b. device connection just at the beginning

The implementation changes based on that requirement, but it's a moot point now since we're all on board with A. Thanks for the clarifications!